src/runtime/current_thread/task_executor.rs

```diff
@@ -24,34 +26,73 @@ impl TaskExecutor {
     ///
     /// For more details, see the [module level](index.html) documentation.
     pub fn current() -> TaskExecutor {
-        TaskExecutor {
-            inner: TaskExecutor02::current(),
-        }
+        TaskExecutor { _p: () }
     }
 
     /// Spawn a `futures` 0.1 future onto the current `CurrentThread` instance.
     pub fn spawn_local(
         &mut self,
-        future: impl Future01<Item = (), Error = ()> + 'static,
+        future: Box<dyn Future01<Item = (), Error = ()>>,
```

Comment:
why not box internally here?

Reply:
IIRC the answer is "because i copied some fn prototypes from the tokio 0.1 version"

Reply:
ah ok

Reply:
as far as drop-in compatibility goes i think code that _calls_ this will still work identically but i am unsure if there are other weird edge cases where it needs to have the _exact_ same prototype?